libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/util/CacheResultUtils.kt

```diff
@@ -51,6 +54,155 @@ internal object CacheResultUtils {
         }
     }
 
+    data class CacheResultKeyRouteTraffic<R>(
+        val route: NavigationRoute,
+        val trafficProvider: (RouteLeg) -> List<String>?
+    ) :
+        CacheResultCall<(NavigationRoute, (RouteLeg) -> List<String>?) -> R, R> {
+
+        override fun invoke(f: (NavigationRoute, (RouteLeg) -> List<String>?) -> R) = f(
+            route,
+            trafficProvider
+        )
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (javaClass != other?.javaClass) return false
+
+            other as CacheResultKeyRouteTraffic<*>
+
+            if (route.id != other.route.id) return false
+            if (trafficProvider != other.trafficProvider) return false
```

Comment:
It will return false when lambdas are different objects (even if their behaviour is the same). Is this expected?

Reply:
Yes, the object is part of the cache key in the current implementation. This is maintained here.

Reply:
Hm, I wonder if that's by design.